packages/ra-core/src/form/useFormGroup.ts

```diff
@@ -67,24 +67,33 @@ export const useFormGroup = (name: string): FormGroupState => {
         valid: true,
     });
 
+    const updateGroupState = useCallback(() => {
+        const fields = formContext.getGroupFields(name);
+        const fieldStates = fields
+            .map(field => {
+                return form.getFieldState(field);
+            })
+            .filter(fieldState => fieldState != undefined); // eslint-disable-line
+
+        const newState = getFormGroupState(fieldStates);
+        setState(oldState => {
+            if (!isEqual(oldState, newState)) {
+                return newState;
+            }
+
+            return oldState;
+        });
+    }, [form, formContext, name]);
+
     useEffect(() => {
+        // Whenever the group content changes (input are added or removed)
+        // we must update its state
+        formContext.subscribe(name, () => {
```

Comment:
subscribe returns an unsubscribe method that you never call. I suspect a memory leak here